### DISCUSSION TITLE
flopy MF6, how to use saverecord as "STEP" or "FREQUENCY" in with fp.mf6.ModflowGwfoc()

### DISCUSSION THREAD

**Discussion by valDallalba** [modflowpy/flopy#1570] [General]
**Describe the bug**

I encounter an issue when trying to save head or budget outputs for a defined __time_step__ array or at a defined __frequency__ when using the __flopy.mf6.ModflowGwfoc()__ method.  I'm using the flopy version : '3.3.6'.

I tried for a simple  simulation, composed of two periods as follow : 

```python
tds = flopy.mf6.ModflowTdis(sim,  pname="tdis",   nper=2,   perioddata=[(1,1, 1.0), (100.0, 100, 1.0)])
```
The idea is to not record at each time step of the second period but only on some defined time step. 

I did not find any tutorial or example using this method to save the output at a defined time step or at a define frequency. This seems to previously work with the method __flopy.modflow.ModflowOc()__ where a dictionary of  __stress_period_data__ could be provided. However,  when using the __flopy.mf6.ModflowGwfoc()__ and defining my __saverecord__ as "STEP" or "FREQUENCY" it seems that it is missing an argument or a keyword to specify the list of time steps or the frequency integer for which to save the head output.

My code looks like this :

```python
oc = flopy.mf6.ModflowGwfoc(gwf, head_filerecord=head_path,  budget_filerecord=budget_path,  saverecord=[['head','step']])
```
And this is the returned error message : 

```python
KeyError: 'steprecord'
```

I would like to know if the help file of the method is missing an argument description or if I miss-understand the way to create the __saverecord__ argument. 

I have tried to add different arguments to the function or to create the  __saverecord__ argument in different ways, with for example an array instead of the keyword, but nothing seems to work for now.  If anybody has some idea for how to specify the list of the time step for which to record the output that would be really helpful. 

Cheers!

Val

**Desktop (please complete the following information):**

 - OS: Ubuntu 18.04.5 LTS
 - Browser : Firefox
 - Version : Firefox -> 105.0 (64 bits), flopy-> 3.3.6

**Comment by christianlangevin** [comment]
Here is an example of a valid saverecord, which you can pass to ModflowGwfoc.  In this case you pass a dictionary where the key is the stress period number.

```
saverecord={0: [("HEAD", "FREQUENCY", 5), ("BUDGET", "FREQUENCY", 5)]}
```

It might also take:

```
saverecord=[("HEAD", "FREQUENCY", 5), ("BUDGET", "FREQUENCY", 5)]
```

I think STEP should work similarly.

**Reply by valDallalba** [reply]
Thank you it works perfectly.

Maybe just for other people that might encounter the problem, the syntax to get multiple time steps outputs is the following (which, at first I found a bit strange) : 

`saverecord=[("HEAD", "STEPS", 1, 5, 6, 7, 10)] `

Thanks!

**Comment by christianlangevin** [comment]
I believe you can pass multiple time step values when using step.  I think the syntax is something like this:

saverecord=[("HEAD", "STEPS", 1, 3, 4, 9), ("BUDGET", "FREQUENCY", 5)]
